dev/src/bulk-writer.ts

```diff
@@ -645,10 +645,12 @@ export class BulkWriter {
    * @param {SetOptions=} options An object to configure the set behavior.
    * @param {boolean=} options.merge - If true, set() merges the values
    * specified in its data argument. Fields omitted from this set() call remain
-   * untouched.
+   * untouched. If your input sets any field to an empty map, all nested fields
+   * at this location are overwritten.
    * @param {Array.<string|FieldPath>=} options.mergeFields - If provided, set()
    * only replaces the specified field paths. Any field path that is not
-   * specified is ignored and remains untouched.
+   * specified is ignored and remains untouched. If your input sets any field to
+   * an empty map, all nested fields at this location are overwritten.
```

Comment:
I was initially confused since I inferred "location" to mean the document being set. The ambiguous interpretations:
- all nested fields in the document location are overwritten with empty map, but top level fields remain
- if the specified field has a nested field, it's deleted, but if the specified field maps to a value, it's not.

alternative suggestion: "if your input sets any field to an empty map, the specified field is (overwritten/set to) an empty map.

Reply:
You are right, this is a bit confusing. I don't like how your example uses "empty map" twice, but I think we can achieve the same by just removing "at this location". What do you think?

Reply:
Yeah, that works!